tests/unit/test_files_checks.py

```diff
@@ -0,0 +1,265 @@
+from together.utils.files import check_file
+
+
+def test_check_jsonl_valid_general(tmp_path: Path):
+    # Create a valid JSONL file
+    file = tmp_path / "valid.jsonl"
+    content = [{"text": "Hello, world!"}, {"text": "How are you?"}]
+    with file.open("w") as f:
+        f.write("\n".join(json.dumps(item) for item in content))
+
+    report = check_file(file)
+
+    assert report["is_check_passed"]
+    assert report["utf8"]
+    assert report["num_samples"] == len(content)
+    assert report["has_min_samples"]
+
+
+def test_check_jsonl_valid_instruction(tmp_path: Path):
+    # Create a valid JSONL file with instruction format
+    file = tmp_path / "valid_instruction.jsonl"
+    content = [
+        {"prompt": "Translate the following sentence.", "completion": "Hello, world!"},
+        {
+            "prompt": "Summarize the text.",
+            "completion": "Weyland-Yutani Corporation creates advanced AI.",
+        },
+    ]
+    with file.open("w") as f:
+        f.write("\n".join(json.dumps(item) for item in content))
+
+    report = check_file(file)
+
+    assert report["is_check_passed"]
+    assert report["utf8"]
+    assert report["num_samples"] == len(content)
+    assert report["has_min_samples"]
+
+
+def test_check_jsonl_valid_conversational_single_turn(tmp_path: Path):
+    # Create a valid JSONL file with conversational format and 1 user-assistant turn pair
+    file = tmp_path / "valid_conversational_single_turn.jsonl"
+    content = [
+        {
+            "messages": [
+                {"role": "user", "content": "Hello"},
+                {"role": "assistant", "content": "Hi there!"},
+            ]
+        },
+        {
+            "messages": [
+                {"role": "user", "content": "How are you?"},
+                {"role": "assistant", "content": "I am fine."},
+            ]
+        },
+    ]
+    with file.open("w") as f:
+        f.write("\n".join(json.dumps(item) for item in content))
+
+    report = check_file(file)
+
+    assert report["is_check_passed"]
+    assert report["utf8"]
+    assert report["num_samples"] == len(content)
+    assert report["has_min_samples"]
+
+
+def test_check_jsonl_valid_conversational_multiple_turns(tmp_path: Path):
+    # Create a valid JSONL file with conversational format and multiple user-assistant turn pairs
+    file = tmp_path / "valid_conversational_multiple_turns.jsonl"
+    content = [
+        {
+            "messages": [
+                {"role": "user", "content": "Is it going to rain today?"},
+                {
+                    "role": "assistant",
+                    "content": "Yes, expect showers in the afternoon.",
+                },
+                {"role": "user", "content": "What is the weather like in Tokyo?"},
+                {"role": "assistant", "content": "It is sunny with a chance of rain."},
+            ]
+        },
+        {
+            "messages": [
+                {"role": "user", "content": "Who won the game last night?"},
```

Comment:
Can we add a `system` message at the beginning of this conversation to ensure that these three roles are handled properly?